src/main/java/org/kaazing/netx/ws/internal/WsURLConnectionImpl.java

```diff
@@ -156,31 +156,36 @@ public void close(int code) throws IOException {
 
     @Override
     public void close(int code, String reason) throws IOException {
+        byte[] reasonBytes = null;
+
         if (code != 0) {
             //verify code and reason against RFC 6455
             //if code is present, it must equal to 1000 or in range 3000 to 4999
             if (code != 1000 && (code < 3000 || code > 4999)) {
-                    throw new IllegalArgumentException("code must equal to 1000 or in range 3000 to 4999");
+                    throw new IOException("code must equal to 1000 or in range 3000 to 4999");
             }
 
             //if reason is present, it must not be longer than 123 bytes
             if (reason != null && reason.length() > 0) {
                 //convert reason to UTF8 string
                 try {
-                    byte[] reasonBytes = reason.getBytes("UTF8");
-                    if (reasonBytes.length > 123) {
-                        throw new IllegalArgumentException("Reason is longer than 123 bytes");
-                    }
+                    reasonBytes = reason.getBytes("UTF8");
```

Comment:
Recommend hoisting `Charset` into a constant.
```java
private static final Charset UTF_8 = Charset.forName("UTF-8");
```
and using it here instead of `"UTF8"`.
```java
reasonBytes = reason.getBytes(UTF_8);
```